providers/softlayer/src/test/java/org/jclouds/softlayer/features/LoadBalancerApiLiveTest.java

```diff
@@ -0,0 +1,529 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.softlayer.features;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import org.jclouds.softlayer.domain.Datacenter;
+import org.jclouds.softlayer.domain.LoadBalancer;
+import org.jclouds.softlayer.domain.LoadBalancerMember;
+import org.jclouds.softlayer.domain.LoadBalancerStatistics;
+import org.jclouds.softlayer.domain.LoadBalancerHealthMonitor;
+import org.jclouds.softlayer.domain.LoadBalancerListener;
+import org.jclouds.softlayer.domain.LoadBalancerMemberPoolHealth;
+import org.jclouds.softlayer.domain.LoadBalancerServicePrice;
+import org.jclouds.softlayer.domain.LoadBalancerProtocolConfiguration;
+import org.jclouds.softlayer.domain.LoadBalancerServerInstanceInformation;
+import org.jclouds.softlayer.domain.LoadBalancerSubnetId;
+import org.jclouds.softlayer.domain.Location;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.jclouds.util.Predicates2.retry;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests behavior of {@code LoadBalancerApi}
+ */
+@Test(groups = "live")
+public class LoadBalancerApiLiveTest extends BaseSoftLayerApiLiveTest {
+
+   private LoadBalancerApi loadBalancerApi;
+   private LoadBalancer loadBalancer = null;
+   private Datacenter datacenter = null;
+   private LoadBalancerMember loadBalancerMember = null;
+   private LoadBalancerHealthMonitor loadBalancerHealthMonitor = null;
+   private LoadBalancerListener loadBalancerListener = null;
+
+   //load balancer creation config
+   private int subnetId = -1;
+   private int servicePrices[] = {199445, 199465, 205837, 205975};
+   private String testInstancePrivateIp = "";
+
+   @BeforeClass(groups = {"integration", "live"})
+   @Override
+   public void setup() {
+      super.setup();
+
+      loadBalancerApi = api.getLoadBalancerApi();
+
+      datacenter = Iterables.get(api.getDatacenterApi().listDatacenters(), 0);
+      assertNotNull(datacenter, "Datacenter must not be null");
+   }
+
+   @Test
+   public void testCreateLoadBalancer() throws Exception {
+      assertTrue(subnetId != -1, "Please set the subnetId");
+      assertTrue(!testInstancePrivateIp.equals(""), "Please set the testInstancePrivateIp for Member functions testing");
+      loadBalancer = LoadBalancer.create(0, 0, "", new Date(), "", 0, 0,
```

Comment:
The reason for this is because of the wait state at line 113-125, the assignment of the `loadBalancer` object needs to be created (not null) for the `apply` function.